### ISSUE TITLE
Report viewing in no longer working on BIRT 4.13.0 v202304181804

### ISSUE THREAD

**Issue by daverch** [eclipse-birt/birt#1280]
**Not sure why since I have not tested this in about 6 months:**
<html><body>
<h2>HTTP ERROR 500 javax.servlet.ServletException: 
org.apache.jasper.JasperException: org.apache.jasper.JasperException: 
Unable to compile class for JSP</h2>

URI: | /viewer/frameset
-- | --
500
javax.servlet.ServletException:  org.apache.jasper.JasperException: org.apache.jasper.JasperException:  Unable to compile class for JSP
ViewerServlet
javax.servlet.ServletException:  org.apache.jasper.JasperException: org.apache.jasper.JasperException:  Unable to compile class for JSP
org.apache.jasper.JasperException: org.apache.jasper.JasperException: Unable to compile class for JSP
org.apache.jasper.JasperException: Unable to compile class for JSP
javax.el.ELException: Provider com.sun.el.ExpressionFactoryImpl not found
java.lang.ClassNotFoundException: com.sun.el.ExpressionFactoryImpl cannot be found by org.eclipse.jetty.osgi.boot_10.0.11

<h3>Caused by:</h3><pre>javax.servlet.ServletException: org.apache.jasper.JasperException: org.apache.jasper.JasperException: Unable to compile class for JSP
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:157)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:822)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
	at org.eclipse.jetty.server.Server.handle(Server.java:563)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$0(HttpChannel.java:505)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:762)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:497)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:934)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1078)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.jasper.JasperException: org.apache.jasper.JasperException: Unable to compile class for JSP
	at org.apache.jasper.servlet.JspServletWrapper.handleJspException(JspServletWrapper.java:589)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:425)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:379)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:327)
	at org.eclipse.jetty.jsp.JettyJspServlet.service(JettyJspServlet.java:101)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:584)
	at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1419)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1665)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:527)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:596)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1383)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1305)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129)
	at org.eclipse.jetty.server.Dispatcher.include(Dispatcher.java:119)
	at org.eclipse.birt.report.presentation.aggregation.layout.FramesetFragment.service(FramesetFragment.java:89)
	at org.eclipse.birt.report.servlet.ViewerServlet.__doGet(ViewerServlet.java:167)
	at org.eclipse.birt.report.servlet.BirtSoapMessageDispatcherServlet.doGet(BirtSoapMessageDispatcherServlet.java:143)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:497)
	at org.apache.axis.transport.http.AxisServletBase.service(AxisServletBase.java:332)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:584)
	at org.eclipse.birt.report.servlet.BirtSoapMessageDispatcherServlet.service(BirtSoapMessageDispatcherServlet.java:115)
	at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1419)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1665)
	at org.eclipse.birt.report.filter.ViewerFilter.doFilter(ViewerFilter.java:71)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:210)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1635)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:527)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1383)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1305)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:149)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:141)
	... 14 more
Caused by: org.apache.jasper.JasperException: Unable to compile class for JSP
	at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:621)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:399)
	... 65 more
Caused by: javax.el.ELException: Provider com.sun.el.ExpressionFactoryImpl not found
	at javax.el.FactoryFinder.newInstance(FactoryFinder.java:60)
	at javax.el.FactoryFinder.find(FactoryFinder.java:141)
	at javax.el.ExpressionFactory.newInstance(ExpressionFactory.java:140)
	at javax.el.ExpressionFactory.newInstance(ExpressionFactory.java:110)
	at org.apache.jasper.compiler.PageInfo.&lt;init&gt;(PageInfo.java:80)
	at org.apache.jasper.compiler.Compiler.generateJava(Compiler.java:120)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:391)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:367)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:351)
	at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:605)
	... 66 more
Caused by: java.lang.ClassNotFoundException: com.sun.el.ExpressionFactoryImpl cannot be found by org.eclipse.jetty.osgi.boot_10.0.11
	at org.eclipse.osgi.internal.loader.BundleLoader.generateException(BundleLoader.java:541)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:536)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:416)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:168)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at org.eclipse.jetty.webapp.WebAppClassLoader.loadClass(WebAppClassLoader.java:511)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at javax.el.FactoryFinder.newInstance(FactoryFinder.java:44)
	... 75 more
</pre>
</body>
</html>

**Comment by speckyspooky** [comment]
I have retested the latest versions of 4.13.0-20230418 for Windows, RCP-designer & All-In-One-designer with the integrated JDK.
My testcases eclipse viewer, standard preview, HTML-preview and PDF in both bundles and I didn't get your error. all works fine

Which bundle do you use, which JDK was used integrated or external (version?)?

My tested versions:

[birt-rcp-report-designer-4.13.0-20230418-win32.win32.x86_64.zip](https://www.eclipse.org/downloads/download.php?file=/birt/downloads/drops/snapshot/birt-rcp-report-designer-4.13.0-20230418-win32.win32.x86_64.zip)

[birt-report-designer-all-in-one-4.13.0-20230418-win32.win32.x86_64.zip](https://www.eclipse.org/downloads/download.php?file=/birt/downloads/drops/snapshot/birt-report-designer-all-in-one-4.13.0-20230418-win32.win32.x86_64.zip)

**Comment by daverch** [comment]
Unfortunately, I cannot use either one of those releases because they are stripped of the perspectives I use. I am using standard EclipseIDE Version: 2023-03 (4.27.0)
Build id: 20230309-1520 with BIRT. I downloaded [Eclipse IDE for Enterprise Java and Web Developers](https://www.eclipse.org/downloads/packages/release/2023-03/r/eclipse-ide-enterprise-java-and-web-developers)

**Comment by speckyspooky** [comment]
The latest eclipse version is currently not merged in full with the latest eclipse version 2023-03.
@wimjongman have you a plan here for a first drafting version of the integration of 2023-03?

**Comment by wimjongman** [comment]
I do, but I am fighting the target platform.

**Comment by merks** [comment]
This seems to be the same problem as this one:

https://github.com/eclipse-birt/birt/issues/1301

Which should be fixed by this PR:

https://github.com/eclipse-birt/birt/pull/1302

**Comment by Ricorder** [comment]
> Missing dependency was added.

Hi, how I can add missing dependency?
I am a junior specialist and it is not easy for me to understand

**Comment by merks** [comment]
@Ricorder 

You shouldn't need to do anything.  The latest build should have no missing dependencies.

**Comment by Ricorder** [comment]
> @Ricorder
> 
> You shouldn't need to do anything. The latest build should have no missing dependencies.

But I get the same error

**Comment by speckyspooky** [comment]
Which version due you try to start and which installation details?
Have you this issue on designer preview or integrated with your application server?

The latest retest after the fix was ok.

**Comment by Ricorder** [comment]
I use version: 2023-06 (4.28.0)
This issue on designer preview

**Comment by speckyspooky** [comment]
Please try the latest built from download page and let us know.
Because I doesn't know which version you use.
On the download page you have a very detailed version number.
And the latest built is from last weekend.

**Comment by Ricorder** [comment]
I just installed the latest version of eclipse and via the marketplace birt, but nothing has changed

Eclipse IDE for Java Developers (includes Incubating components)
Version: 2023-06 (4.28.0)
Build id: 20230608-1333

**Comment by speckyspooky** [comment]
Have you downloaded from here? 
https://download.eclipse.org/birt/updates/nightly/latest/index.html

**Comment by merks** [comment]
The marketplace listing is installing from https://download.eclipse.org/birt/updates/release/latest which is not the 4.14 version with the fix.   I don't think the 4.14 update site 

https://download.eclipse.org/birt/updates/nightly/latest

will install into 2023-06 (4.28) because BIRT builds against and includes features from 2023-09 (4.29) (which is a questionable practice but not the only one). 

I think 4.14 will install into one of these:

https://www.eclipse.org/downloads/packages/release/2023-09/m1

But of course easier to grab the package on the nightly page's downloads section.

**Comment by Ricorder** [comment]
> Have you downloaded from here?
> https://download.eclipse.org/birt/updates/nightly/latest/index.html

I have installed this birt-rcp-report-designer-4.14.0-202307151849-macosx.cocoa.x86_64.zip 
from https://download.eclipse.org/birt/updates/nightly/latest/index.html
but it didn't help

**Comment by Ricorder** [comment]
> The marketplace listing is installing from https://download.eclipse.org/birt/updates/release/latest which is not the 4.14 version with the fix. I don't think the 4.14 update site
> 
> https://download.eclipse.org/birt/updates/nightly/latest
> 
> will install into 2023-06 (4.28) because BIRT builds against and includes features from 2023-09 (4.29) (which is a questionable practice but not the only one).
> 
> I think 4.14 will install into one of these:
> 
> https://www.eclipse.org/downloads/packages/release/2023-09/m1
> 
> But of course easier to grab the package on the nightly page's downloads section.

I tried, but the result is the same

**Comment by Ricorder** [comment]
<img width="774" alt="Снимок экрана 2023-07-22 в 13 55 36" src="https://github.com/eclipse-birt/birt/assets/48915901/9ba9d2b4-d2ef-4120-ba36-361a1af324f1">

**Comment by merks** [comment]
Please show the Help -> About dialog's Installation Details page.  Ensure that it's big enough to show all the installed features and their versions.   Note too that using the Marketplace will **NOT** install a version with the fix.

Try one of these (whatever is appropriate for your hardware):

- https://www.eclipse.org/downloads/download.php?file=/birt/updates/nightly/N202307151849/downloads/birt-report-designer-all-in-one-4.14.0-202307151849-macosx.cocoa.aarch64.zip
- https://www.eclipse.org/downloads/download.php?file=/birt/updates/nightly/N202307151849/downloads/birt-report-designer-all-in-one-4.14.0-202307151849-macosx.cocoa.x86_64.zip

**Comment by speckyspooky** [comment]
I started a retest of the "All-in-One"-designer & "RCP"-designer (Windows environment)
and the download according the BIRT-donload-page (not marketplace) of latest built BIRT 4.14.0 - see below.

For testing I used my latest bugfix-reports and the reports were created.
The preview is available without any issues and I was also able to create pdf-reports on it.

So I cannot find an issue with dependencies currently on the Windows-version.

![BIRT 4 14 0 Nightly - N202307151849 – Mozilla Firefox](https://github.com/eclipse-birt/birt/assets/41593722/a22d72b5-3b33-4808-b785-824c90dce18f)
